src/js/components/CollapsibleErrorMessage.js

```diff
@@ -0,0 +1,110 @@
+
+import Icon from './Icon';
+
+import React from 'react';
+import classNames from 'classnames';
+
+/**
+ * An error bar component commonly used in the ServiceForm and the JobsForm
+ * that contains collapsible error messages.
+ */
+class CollapsibleErrorMessage extends React.Component {
+
+  /**
+   * Initialize superclass
+   */
+  constructor() {
+    super(...arguments);
+
+    // Initial state
+    this.state = {
+      expanded: false
+    };
+
+  }
+
+  /**
+   * Toggle the expanded state of the collapsible error bar
+   */
+  toggleExpanded() {
+    this.setState({
+      expanded: !this.state.expanded
+    });
+  }
+
+  /**
+   * React.js Render Function
+   *
+   * @returns {React.Component} - The rendered content
+   */
+  render() {
+    let {message, details} = this.props;
+    let isVisible = !!message;
+    let isDetailed = isVisible && details && details.length;
+    let contents = [];
+
+    // For sure, create a message body
+    contents.push(
+        <div key="fixed" className="collapsible-fixed text-align-center text-danger flush-top">
+          <Icon
+            family="mini"
+            id="yield"
+            size="mini"
+            className="icon-alert icon-margin-right"
+            color="red" />
+          {message}
+          {(() => {
+            if (isDetailed) {
+              return (
+                  <span>
+                    &nbsp;
```

Comment:
I think it's better to add a whitespace after the message. More than one whitespace won't appear in HTML either ways ;)